client-core/src/service/wallet_service.rs

```diff
@@ -307,31 +485,72 @@ where
             ));
         }
 
-        self.set_wallet(name, enckey, Wallet::new(view_key))
+        self.set_wallet(name, enckey, Wallet::new(view_key.clone()))?;
+
+        let info_keyspace = get_info_keyspace(name);
+        // key: "viewkey"
+        // value: view-key
+        write_pubkey_enc(&self.storage, &info_keyspace, "viewkey", &view_key, enckey)?;
+
+        // key: index
+        // value: walletname
+        let wallet_keyspace = get_wallet_keyspace();
+        self.storage
+            .set(wallet_keyspace, name, name.as_bytes().to_vec())?;
+
+        Ok(())
     }
 
     /// Returns view key of wallet
     pub fn view_key(&self, name: &str, enckey: &SecKey) -> Result<PublicKey> {
-        let wallet = self.get_wallet(name, enckey)?;
-        Ok(wallet.view_key)
+        let _wallet_found = self.get_wallet(name, enckey)?;
```

Comment:
this is just to check if the wallet exists?

Reply:
yes, because integration needs exact the same error message